src/java/org/apache/cassandra/index/IndexStatusManager.java

```diff
@@ -84,26 +85,39 @@ private IndexStatusManager()
      */
     public <E extends Endpoints<E>> E filterForQuery(E liveEndpoints, Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel level)
     {
+        int required = level.blockFor(keyspace.getReplicationStrategy());
+        ReplicaCollection.Builder<E> buildSucceededEndpoints = liveEndpoints.newBuilder(0);
```

Comment:
```suggestion
    ReplicaCollection.Builder<E> buildSucceededEndpoints = liveEndpoints.newBuilder(liveEndpoints.size());
```
The only other option would be to use `required` as an initial size, but I think we'll expect most nodes to be in the `BUILD_SUCCEEDED` state most of the time, so that would force more array creations. For instance, if we have an RF = 3, and CL = QUORUM, we probably want to use `liveEndpoints.size()` which would be 3, instead of using 2 and having to do an extra array copy if all three replicas are `BUILD_SUCCEEDED`.

WDYT?

Reply:
I'm no expert, but what you suggested makes sense. I'll use `liveEndpoints.size()`.